src/chat/chat.controller.ts

```diff
@@ -54,6 +63,106 @@ export class ChatController {
   @UseGuards(JwtAuthGuard)
   @Get('room/:roomId/members')
   async members(@Param('roomId') roomId: string) {
-    return this.gateway.getRoomMembers(roomId);
+    return this.chatGateway.getRoomMembers(roomId);
+  }
+
+  // 방 들어가기
+  @UseGuards(JwtAuthGuard)
+  @Post('room/:roomId/join')
+  async join(@ReqUser() user: JwtPayloadDto, @Param('roomId') roomId: string) {
+    const profileId = user.sub;
+    const exists = await this.userService.findByProfileId(profileId);
+    if (!exists) {
+      throw new UnauthorizedException('유효하지 않는 사용자');
+    }
+
+    // 방 있는지 확인하기
+    const room = await this.chatService.findRoomById(roomId);
+    if (!room) throw new NotFoundException('room not found');
+
+    // Participants 업데이트(있으면 패스)
+    const already = await this.chatService.isParticipant(roomId, profileId);
+    if (!already) {
+      await this.chatService.addParticipant(roomId, profileId);
+    }
+
+    //소켓 join
+    await this.chatGateway.joinProfileToRoom(profileId, roomId);
+
+    return {
+      roomId,
+      alreadyParticipant: already, // true면 기존 멤버였음, false는 신규 입장
+    };
+  }
+
+  // 전체 채팅방 가져오기
+  @UseGuards(JwtAuthGuard)
+  @Get('rooms')
+  async getRooms(@ReqUser() user: JwtPayloadDto) {
+    const profileId = user.sub;
+    const exists = await this.userService.findByProfileId(profileId);
+    if (!exists) {
+      throw new UnauthorizedException('유효하지 않는 사용자');
+    }
+  }
```

Comment:
`profileId`를 인자로만 사용한다면 굳이 선언해줄 필요는 없을 것 같습니다.
```suggestion
  async getRooms(@ReqUser() user: JwtPayloadDto) {
    const exists = await this.userService.findByProfileId(user.sub);
    if (!exists) {
      throw new UnauthorizedException('유효하지 않는 사용자');
    }
  }
```

Reply:
단순 변수 선언 부분에 대해서 수정 작업 하여 반영하였습니다.